src/modules/roc_core/target_linux/roc_core/timer.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/target_linux/roc_core/timer.h
+//! @brief Thread-safe timer.
+
+#ifndef ROC_CORE_TIMER_H_
+#define ROC_CORE_TIMER_H_
+
+#include <sys/timerfd.h>
+
+#include "roc_core/atomic.h"
+#include "roc_core/noncopyable.h"
+#include "roc_core/seqlock.h"
+#include "roc_core/time.h"
+
+namespace roc {
+namespace core {
+
+//! Thread-safe timer.
+class Timer : public NonCopyable<> {
+public:
+    Timer();
+
+    virtual ~Timer();
+
+    //! Set timer deadline.
+    //! Can be called concurrently, but only one concurrent call will succeed.
+    //! Returns false if the call failed because of another concurrent call.
+    //! Is lock-free if Semaphore::post() is so (which is true of modern plarforms).
+    //! Current or future wait_deadline() call will unblock when deadline expires.
+    //! Zero deadline means wake up immediately.
+    //! Nagative deadline means never wake up, until deadline is changed again.
+    bool try_set_deadline(nanoseconds_t deadline);
+
+    //! Wait until deadline expires.
+    //! Should be called from a single thread.
+    //! Assumes that wait_deadline() calls are serialized.
+    //! Deadline may be changed concurrently from other threads.
+    void wait_deadline();
+
+private:
+    struct itimerspec convert_deadline();
+    struct itimerspec infinity_deadline();
+    struct itimerspec immediately_deadline();
+    struct itimerspec finity_deadline();
```

Comment:
Private methods should end with underscore. However I'd suggest to make them free functions (private to .cpp file, in anonymous namespace) instead of methods.

Reply:
ok